### PR TITLE
[editing] Integrate evil-vimish-fold via dotspacemacs-folding-method

### PR DESCRIPTION
**Pull request by Kazark**
Not confident that I have plugged this in everywhere that I should have. However
I tested locally and it seems to be working. Feedback coveted.

### CODE REVIEW

File: layers/+spacemacs/spacemacs-editing/packages.el [hunk 1]
```diff
@@ -31,6 +31,8 @@
         string-inflection
         undo-tree
         uuidgen
+        (vimish-fold :toggle (eq 'vimish dotspacemacs-folding-method))
+        (evil-vimish-fold :toggle (eq 'vimish dotspacemacs-folding-method))
```
Comment (robbyoconnor) [inline layers/+spacemacs/spacemacs-editing/packages.el:35]: use `when`.

**Review by robbyoconnor — CHANGES_REQUESTED** [review]

File: layers/+spacemacs/spacemacs-editing/packages.el (hunk 1 above)
Comment (robbyoconnor) [reply layers/+spacemacs/spacemacs-editing/packages.el:35]: https://www.gnu.org/software/emacs/manual/html_node/elisp/Conditionals.html

Comment (Kazark) [reply layers/+spacemacs/spacemacs-editing/packages.el:35]: @robbyoconnor like this?

```elisp
(when (eq 'vimish dotspacemacs-folding-method) vimish-fold evil-vimish-fold)
```

Hm, that can't be quite right.

Comment (Kazark) [reply layers/+spacemacs/spacemacs-editing/packages.el:35]: Do I need to use a quasiquote as well?

```elisp
`(
  ...
  @,(when (eq 'vimish dotspacemacs-folding-method) '(vimish-fold evil-vimish-fold))
  ...
  )
```

Comment (robbyoconnor) [reply layers/+spacemacs/spacemacs-editing/packages.el:35]: Check other layers

Comment (smile13241324) [reply layers/+spacemacs/spacemacs-editing/packages.el:35]: I don't think we need a `when` statement here, as `when` and `if` take a condition and then either execute an action or choose between two alternative actions. In this case this is already covered by the `:toggle` label which will take the condition.

**Review by smile13241324 — APPROVED** [review]
Looks good so far, I think you have found all the places to change.

**Comment by smile13241324** [comment]
Thank you for contributing to spacemacs 💜, I have cherry picked your commit into develop.

**Comment by ECon87** [comment]
New to emacs so probably an error on my part. I am on the develop branch but I can't seem to get this to work.

I am working on an R file; I enclose a snippet of code witin "# {{{" and "# }}}", however nothing gets folded after I press "z c".

In python only the line underneath gets folded (instead of the whole snippet of code).

Thanks.

**Comment by lebensterben** [comment]
@ECon87 Can you post a minimum example R script so I can try to reproduce the bug?

**Comment by ECon87** [comment]
> @ECon87 Can you post a minimum example R script so I can try to reproduce the bug?

```
library(dplyr)

# ==================
# Directory Paths
# ==================

# {{{

Root0Dir <- "some/path/"
TabFigDir <- paste0(Root0Dir, "Tables_Figures/")
RootDir <- paste0(Root0Dir,"Empirical/")
DataDir <- paste0(RootDir, "Data/")


# }}}

```

The above snippet should it. When I press "z c" while it folds the area above "# {{{", and not below. If I remove the comment, then "zc" does not fold anything.

Thanks for the response.

**Comment by lebensterben** [comment]
I can confirm this.
But I'm not sure this is bug.
Neither can Vim fold it.

Do you know any particular editor that folds them?

**Comment by ECon87** [comment]
I can fold it in Vim. But just to make sure we are talking about identical items.

I have the following code in my vimrc

```
autocmd BufNewFile,BufRead *.r,*.R :set foldmethod=marker
```

Then, when I press "z R" (or z o, or z r or z a and so). The fold works as expected.

**Comment by lebensterben** [comment]
@ECon87 

https://github.com/alexmurray/evil-vimish-fold

This may work. In particular it folds {{{ }}}.

**Comment by ECon87** [comment]
I might be wrong, but isn't this already working underneath the spacemacs fold functions?

**Comment by Kazark** [comment]
@ECon87

1. What is `dotspacemacs-folding-method` set to in your `.spacemacs`?
2. Do you have this problem in other major modes?

**Comment by ECon87** [comment]
Hey @Kazark,

1. It's set to 'evil

2. In Python too. I am only using R and Python to be honest, and i am new to emacs.

**Comment by Kazark** [comment]
@ECon87 this will definitely not work with it set to `'evil`. This is a bit confusing, but there is no folding package for Emacs that provides the full Vim functionality out of the box, so what this PR did was it gave the `'vimish` option, which uses the package that @lebensterben linked to. Trying setting `dotspacemacs-folding-method 'vimish` and reloading your configuration (`SPC f e R`; you may also need to restart Spacemacs, not sure; `SPC q r`), then let us know if it's still not working.

**Comment by ECon87** [comment]
@Kazark that worked like a charm. Thank you.

I checked the merged code above, and just noticed that the possible values for the `dotspacemacs-folding-method` are `evil, origami, and vimish`.  My `.spacemacs` only listed `evil and origami` as possible values.

For anyone else on spacemacs develop branch, you might need to update via `cd .emacs.d; pull --rebase`.

**Comment by Kazark** [comment]
@ECon87 you may need to learn about `SPC f e D` (diff your `.spacemacs` with the newest version).

**Pull request merged**